src/Microsoft.AspNetCore.Authentication/ClaimsTransformationContext.cs

```diff
@@ -0,0 +1,11 @@
+﻿using Microsoft.AspNetCore.Http;
+using System.Security.Claims;
+
+namespace Microsoft.AspNetCore.Authentication
+{
+    public class ClaimsTransformationContext
+    {
+        public HttpContext Context { get; set; }
+        public ClaimsPrincipal Principal { get; set; }
```

Comment:
Setter seems fine for Principal in case of wanting to replace it.  I think its less reasonable for replacing the HttpContext, so make that readonly/ctor param